tests/test_replicate.py

```diff
@@ -0,0 +1,187 @@
+import subprocess
+
+# Python 2 and 3 compatible
+try:
+    from unittest.mock import MagicMock
+    from unittest.mock import patch
+except ImportError:
+    from mock import MagicMock
+    from mock import patch
+
+import boto3
+import pytest
+import google
+import scripts.aws_replicate
+from scripts.aws_replicate import AWSBucketReplication, exec_aws_copy
+from scripts.google_replicate import exec_google_copy, resumable_streaming_copy
+
+from scripts.errors import APIError
+
+TEST_UUID = "11111111111111111111111111111111"
+TEST_FILENAME = "test"
+TEST_URL = ["test_url1", "test_url2"]
+
+
+def gen_mock_manifest_data():
+    fake = {
+        "id": "11111111111111111",
+        "filename": "abc.bam",
+        "size": 1,
+        "md5": "1223344543t34mt43tb43ofh",
+        "acl": "tgca",
+        "project_id": "TGCA",
+    }
+
+    return [
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+        fake,
+    ]
+
+
+@patch("google.cloud.storage.Client")
+@patch("scripts.google_replicate.check_blob_name_exists_and_match_md5_size")
+@patch("scripts.google_replicate.bucket_exists")
+def test_resumable_streaming_copy_called(
+    mock_bucket_exists, mock_check_blob, mock_client
+):
```

Comment:
I would add some docstrings for these tests to clarify exactly what you're testing

Reply:
fixed in d47a36c889171c98bbe27a87e89fb6d55d356f0f